EIPS/eip-2678.md

```diff
@@ -0,0 +1,2132 @@
+including source code and any/all deployed instances across multiple
+networks. Package manifests are minified JSON objects, to be distributed
+via content addressable storage networks, such as IPFS.
+
+This document presents a natural language description of a formal
+specification for version **3** of this format.
+
+
+Motivation
+==========
+
+This standard aims to encourage the Ethereum development ecosystem
+towards software best practices around code reuse. By defining an open,
+community-driven package data format standard, this effort seeks to
+provide support for package management tools development by offering a
+general-purpose solution that has been designed with observed common
+practices in mind.
+
+As version 3 of this specification, this standard seeks to address a
+number of areas of improvement found for the previous version (defined
+in
+[EIP-1123](https://eips.ethereum.org/EIPS/eip-1123)).
+This version:
+
+
+-   Updates the schema for a *package manifest* to be compatible with
+	the [metadata](https://solidity.readthedocs.io/en/latest/metadata.html) output for compilers.
+-   Updates the `"sources"` object definition to support a wider range of source file types and serve as [JSON input](https://solidity.readthedocs.io/en/latest/using-the-compiler.html#compiler-input-and-output-json-description) for a compiler.
+-   Moves compiler definitions to a top-level `"compilers"` array to:
+	-   Simplify the links between a compiler version, sources, and its
+		compiled assets.
+	-   Simplify packages that contain multiple compiler versions.
+-   Updates key formatting from `snake_case` to `camelCase` to be
+	more consistent with [JSON convention](https://google.github.io/styleguide/jsoncstyleguide.xml?showone=Property_Name_Format#Property_Name_Format).
+
+<div id="package-specification"></div>
+
+Specification
+=============
+
+This document defines the specification for an EthPM package manifest. A
+package manifest provides metadata about a *package*, and
+in most cases should provide sufficient information about the packaged
+contracts and its dependencies to do bytecode verification of its
+contracts.
+
+A [hosted version](http://ethpm.github.io/ethpm-spec)
+of this specification is available via GitHub Pages. This EIP and the
+hosted HTML document were both autogenerated from the same documentation
+source.
+
+
+Guiding Principles
+------------------
+
+This specification makes the following assumptions about the document
+lifecycle.
+
+1.  Package manifests are intended to be generated programmatically by
+    package management software as part of the release process.
+
+2.  Package manifests will be consumed by package managers during tasks
+    like installing package dependencies or building and deploying new
+    releases.
+
+3.  Package manifests will typically **not** be stored alongside the
+    source, but rather by package registries *or* referenced by package
+    registries and stored in something akin to IPFS.
+
```

Comment:
Maybe add another item:
"Package manifests can be used to verify public deployments of source contracts" a la Etherscan verification